source/documentation/how-to/ha-vms.html.md

```diff
@@ -0,0 +1,86 @@
+
+## Are highly available VMs always restarted automatically?
+
+Starting a VM automatically is a bit more complicated than one might initially think.  Several conditions must be satisfied in order to start such a VM safely and consistently with user actions.
+
+The very first condition is that oVirt Engine must be fully certain that the VM is not running anywhere at the moment.  Failing to satisfy this condition could lead to multiple instances of the VM running concurrently on different hosts, so called *split brain* situation.  oVirt uses safety mechanisms such as sanlock data, VM leases and resume behaviors to prevent split brain.  But it's still important to be aware that if oVirt Engine doesn't know (under certain setups) for sure that a VM is not already running, it can't start it automatically elsewhere.  For instance, if a host running a VM is unreachable it doesn't mean that the VM is not happily running there.  If corresponding protection mechanisms are not enabled then oVirt Engine may not be able to start the VM elsewhere, despite it is highly available.
+
+VMs manually shut down or powered off from oVirt or from the guest operating system are not restarted automatically.  Switching VMs off manually is a user action and oVirt honors it.  The user is responsible for starting such a VM again manually.
+
+There may be other reasons preventing start of a highly available VM, such as impossibility to run the VM due to resource constraints at a given moment.
+
+## Paused VMs
+
+Besides being running, stopped or unreachable, highly available VMs (as well as other VMs) may also be paused. VMs can be paused for different reasons, for instance:
+
+- A user asks the VM to start paused.
+- A VM can be paused for a short while in the final stages of migrations or when making a snapshot with memory.  This typically doesn't harm VM availability.
+- When a thin provisioned disk image is getting full and must be enlarged.  This is again only temporary situation.
+- An I/O error due to inaccessible or broken storage occurs.  This is a real problem.
+
+A paused VM can be later switched to another state (typically Running or Down) either automatically (e.g. on migrations) or manually by the user.
+
+## Highly available VMs and I/O errors
+
+When a VM attempts to access an unreachable or otherwise broken storage and I/O error occurs, the VM gets paused.  Then it can't get running again until the storage problem disappears.  The paused VMs can then be handled in different ways by oVirt.  Once storage is working again, Vdsm tries to resume all the VMs, highly available or not, on the given host that have been paused due to storage errors (it doesn't try to resume any other paused VMs).  Whether any such VM is actually resumed automatically by Vdsm or handled differently depends on VM *resume behavior*.
```

Comment:
You first write that "Once storage is working again, Vdsm tries to resume all the VMs, highly available or not" and than "Whether any such VM is actually resumed automatically by Vdsm or handled differently depends on VM *resume behavior*". I would propose to write it slightly differently, something like:
"Once storage is working again, VDSM goes over all paused VMs and handles them according to the configured *resume behavior*"